pkg/validation/attributes/attributes.go

```diff
@@ -0,0 +1,62 @@
+package attributes
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	apiAttributes "github.com/devfile/api/v2/pkg/attributes"
+)
+
+// ValidateGlobalAttribute validates the workspace template spec data for global attribute references
+func ValidateGlobalAttribute(workspaceTemplateSpec *v1alpha2.DevWorkspaceTemplateSpec) error {
+
+	var err error
+
+	if workspaceTemplateSpec != nil {
+		// Validate the components
+		if err = ValidateComponents(workspaceTemplateSpec.Attributes, &workspaceTemplateSpec.Components); err != nil {
```

Comment:
I'm not clear on why we're passing references to slices throughout this code -- if the concern is being able to modify the underlying data, passing the slices is sufficient (and cleaner). Is there a reason we need to pass pointers to structs everywhere?

For context, passing `workspaceTemplateSpec.Components` *will* copy the slice struct, but the slice struct is just a bunch of references to memory locations.

Reply:
Addressed